specs/dwritecore/DWriteCore.md

```diff
@@ -0,0 +1,255 @@
+            throw Exception{ HRESULT_FROM_WIN32(GetLastError()) };
+        }
+
+        // Call a GDI function to fill in the DIBSECTION structure for the bitmap.
+        DIBSECTION dib;
+        if (!GetObjectW(dibHandle, sizeof(dib), &dib))
+        {
+            throw Exception{ HRESULT_FROM_WIN32(GetLastError()) };
+        }
+
+        result.width = dib.dsBm.bmWidth;
+        result.height = dib.dsBm.bmHeight;
+        result.pixels = static_cast<uint32_t*>(dib.dsBm.bmBits);
+    }
+
+    return result;
+}
+```
+
+# API Details
+
+```c++
+/// <summary>
+/// Specifies the type of DirectWrite factory object.
+/// DirectWrite factory contains internal state such as font loader registration and cached font data.
+/// In most cases it is recommended to use the shared factory object, because it allows multiple components
+/// that use DirectWrite to share internal DirectWrite state and reduce memory usage.
+/// However, there are cases when it is desirable to reduce the impact of a component,
+/// such as a plug-in from an untrusted source, on the rest of the process by sandboxing and isolating it
+/// from the rest of the process components. In such cases, it is recommended to use an isolated factory 
+/// for the sandboxed component.
+/// </summary>
+enum DWRITE_FACTORY_TYPE
+{
+    /// <summary>
+    /// Shared factory allow for re-use of cached font data across multiple in process components.
+    /// Such factories also take advantage of cross process font caching components for better performance.
+    /// </summary>
+    DWRITE_FACTORY_TYPE_SHARED,
+
+    /// <summary>
+    /// Objects created from an isolated factory do not modify internal state or cached data used by
+    /// objects from other factories.
+    /// </summary>
+    DWRITE_FACTORY_TYPE_ISOLATED,
+
+#if DWRITE_CORE
+    /// <summary>
+    /// Objects created from a restricted factory do not use or modify internal state or cached data
+    /// used by other factories. In addition, the system font collection contains only well-known fonts.
+    /// </summary>
+    DWRITE_FACTORY_TYPE_RESTRICTED
+#endif // DWRITE_CORE
+};
+```
+
+```c++
+#if DWRITE_CORE
+
+/// <summary>
+/// Contains information about a bitmap associated with an IDWriteBitmapRenderTarget.
+/// </summary>
+struct DWRITE_BITMAP_DATA_BGRA32
+{
+    UINT32 width;
+    UINT32 height;
+    _Field_size_(width * height) UINT32* pixels;
+};
+
+/// <summary>
+/// Encapsulates a bitmap which can be used for rendering glyphs.
+/// </summary>
+DWRITE_BEGIN_INTERFACE(IDWriteBitmapRenderTarget2, "C553A742-FC01-44DA-A66E-B8B9ED6C3995") : IDWriteBitmapRenderTarget1
+{
+    /// <summary>
+    /// Gets the demensions and a pointer to the system memory bitmap encapsulated by this
+    /// bitmap render target object. The pointer is owned by the render target object, and
+    /// remains valid as long as the object exists.
+    /// </summary>
+    STDMETHOD(GetBitmapData)(_Out_ DWRITE_BITMAP_DATA_BGRA32* bitmapData) PURE;
```

Comment:
probably want to avoid "PURE". if we want to express this as a pure-virtual method, should just = 0;
unless there is some precedent for "PURE" that I've never seen :)

Reply:
CONSIDER: Remove `PURE` as this is a C++ header already and not using the "regular" "build an interface with macros for C and C++"

Reply:
Agreed it's not really useful for C++, but all the existing DWrite methods use PURE.